src/lib.rs

```diff
@@ -70,10 +70,72 @@ pub static README_PATHS: &[&str; 5] = &[
 
 pub static LICENSE_PATHS: &[&str; 3] = &["LICENSE", "LICENSE.md", "COPYING"];
 
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PackageName {
+    pub namespace: String,
+    pub name: String,
+}
+
+impl PackageName {
+    /// Parses the package name from a `namespace/name` format
+    pub fn parse(s: &str) -> Result<Self, &'static str> {
+        if !s.contains('/') {
+            return Err("no / in package name");
+        }
+        let mut split = s.split('/');
+        let name = split.next().ok_or("no name in package name")?.to_string();
+        let namespace = split
+            .next()
+            .ok_or("no namespace in package name")?
+            .to_string();
```

Comment:
Ideally, we'd implement a proper parser (e.g. a mini recursive descent parser) for this instead of relying on `string.split()`. Otherwise we're going to miss lots of edge cases or mis-parse things.

Reply:
This can be pushed off for a later refactor.